types/luxon/src/datetime.d.ts

```diff
@@ -1616,6 +1619,25 @@ export class DateTime<IsValid extends boolean = DefaultValidity> {
      */
     static fromStringExplain(text: string, fmt: string, options?: DateTimeOptions): ExplainedFormat;
 
+    /**
+     * Build a parser for fmt using the given locale. This parser can be passed to {@link DateTime.fromFormatParser} to a parse a date in this format. This can be used to optimize cases where many dates need to be parsed in a specific format.
+     *
+     * @param fmt - the format the string is expected to be in (see description)
+     * @param options - the Locale options
+     */
+    static buildFormatParser(fmt: string, options?: LocaleOptions): unknown;
+
+    /**
+     * Create a DateTime from an input string and format parser.
+     *
+     * The format parser must have been created with the same locale as this call.
+     *
+     * @param text the string to parse
+     * @param formatParser - parser from {@link DateTime.buildFormatParser}
+     * @param opts options taken by fromFormat()
+     */
+    static fromFormatParser(text: string, formatParser: unknown, opts?: DateTimeOptions): DateTimeMaybeValid;
```

Comment:
Currently TS is happy with:
```ts
fromFormat("", "")
```
which is invalid.

Let's define an opaque type for the `TokenParser`:
```ts
declare const tokenParserBrand: unique symbol;
export interface TokenParser {
  [tokenParserBrand]: void;
}
```
That will make the connection between this parameter and the return type of the builder type-safe, without exposing the shape of this token parser.

Reply:
Thanks, I wasn't sure what to do there. I'll add this type.

Reply:
Slightly modified to pass ESLint:

```typescript
export {};

declare const tokenParserBrand: unique symbol;
export interface TokenParser {
    [tokenParserBrand]: true;
}
```

All other types and interfaces are explicitly `export`ed.